### PR TITLE
chore: add reconcile job for deleted crds

### PR DESCRIPTION
**Pull request by yashmehrotra**
Fixes: https://github.com/flanksource/duty/issues/1184

### CODE REVIEW

File: pkg/jobs/topology/topology_jobs.go [hunk 1]
```diff
@@ -157,3 +157,48 @@ var CleanupDeletedTopologyComponents = &job.Job{
...
+
+		if err := ctx.DB().
+			Select("id", "name", "namespace").
+			Where("source IN (?, ?)", models.SourceCRD, models.SourceTopology).
+			Where(duty.LocalFilter).
+			Find(&topologies).Error; err != nil {
+			return err
+		}
+
+		for _, t := range topologies {
```
Comment (moshloop) [inline pkg/jobs/topology/topology_jobs.go:178]: Should we list all topologies rather than 1 by 1?

Comment (yashmehrotra) [reply pkg/jobs/topology/topology_jobs.go:192]: Done

File: pkg/jobs/topology/topology_jobs.go [hunk 2]
```diff
@@ -157,3 +161,46 @@ var CleanupDeletedTopologyComponents = &job.Job{
...
+		if err != nil {
+			return fmt.Errorf("error creating dynamic client for Topology: %w", err)
+		}
+
+		objs, err := client.List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("error listing Topology kind: %w", err)
+		}
+		k8sIDs := lo.Map(objs.Items, func(obj unstructured.Unstructured, _ int) string { return string(obj.GetUID()) })
+		for _, t := range topologies {
```
Comment (moshloop) [inline pkg/jobs/topology/topology_jobs.go:192]:
```suggestion
		if len(topologies) == 0 {
		   // a misconfiguration or intermittent error could cause unnecessary deletion of all topologies
		   ctx.Warnf("Skipping topology CRD cleanup due to zero topologies returned" 
		}
		for _, t := range topologies {
```

File: pkg/jobs/topology/topology_jobs.go [hunk 3]
```diff
@@ -157,3 +161,46 @@ var CleanupDeletedTopologyComponents = &job.Job{
 		return nil
 	},
 }
+
+var TopologyCRDReconcile = &job.Job{
+	Name:       "TopologyCRDReconcile",
+	Schedule:   "@every 1h",
```
Comment (moshloop) [inline pkg/jobs/topology/topology_jobs.go:167]:
```suggestion
	Schedule:   "@every 8h",
```

File: pkg/jobs/topology/topology_jobs.go [hunk 4]
```diff
@@ -157,3 +161,50 @@ var CleanupDeletedTopologyComponents = &job.Job{
...
+		}
+
+		objs, err := client.List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("error listing Topology kind: %w", err)
+		}
+		k8sIDs := lo.Map(objs.Items, func(obj unstructured.Unstructured, _ int) string { return string(obj.GetUID()) })
+		if len(k8sIDs) == 0 {
+			// a misconfiguration or intermittent error could cause unnecessary deletion of all topologies
+			ctx.Warnf("Skipping topology CRD cleanup due to zero topologies returned")
```
Comment (moshloop) [inline pkg/jobs/topology/topology_jobs.go:194]:
```suggestion
			ctx.Warnf("Skipping topology CRD cleanup due to zero topologies returned")
			return nil
```

**Pull request merged**